src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -257,10 +248,78 @@ public void unsubscribe() {
           }
         }
       }
-      processedRecords = _consumerRecordsProcessor.process(rawRecords);
+
+      xRecords = toXRecords(rawRecords);
+      xRecords = _consumerRecordsProcessor.process(xRecords);
```

Comment:
Hmm, a local variable in this case is just a pointer on stack and literally does not really change anything, right?